netket/operator/_abstract_observable.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+
+from netket.hilbert import AbstractHilbert
+
+
+class AbstractObservable(abc.ABC):
```

Comment:
this concept is very useful indeed and more general than AbstractOperator. However, the current abstract interface does not do much, it just contains hilbert, so I am a bit uncertain whether it warrants creating an abstract class? aren't there other methods that should be part of the abstract interface of an Observable/ that are more characteristic than just containing a Hilbert?